lib/amazonka/test/Main.hs

```diff
@@ -0,0 +1,97 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TypeApplications #-}
+{-# OPTIONS_GHC -Wno-incomplete-uni-patterns #-}
+module Main (main) where
+
+import           Control.Concurrent             (threadDelay)
+import           Control.Exception              (Exception, SomeException(..), throwIO, try)
+import           Data.IORef                     (newIORef, atomicModifyIORef')
+import           Data.Time.Clock                (getCurrentTime, addUTCTime)
+import           Network.HTTP.Client            (HttpException(..))
+import           Network.HTTP.Types.Status      (status500)
+import           Test.Tasty
+import           Test.Tasty.HUnit
+
+-- Amazonka imports
+import           Amazonka                       (SessionToken)
+import           Amazonka.Auth.Background       (fetchAuthInBackground)
+import           Amazonka.Auth.Exception        (AuthError(..))
+import           Amazonka.Core                  (Time(..), ISO8601, Sensitive(..))
+import           Amazonka.Types                 (AuthEnv(..), ServiceError(..))
+
+main :: IO ()
+main = defaultMain tests
+
+tests :: TestTree
+tests = testGroup "Auth background error categorization"
+  [ testCase "UnknownAuthError wraps non-auth exceptions" $ do
+      menv <- mkAuthEnvAction (userError "random failure")
+      -- Note: fetchAuthInBackground immediately calls the provided action once
+      -- to seed the initial AuthEnv (this is the "first" call and succeeds),
+      -- then a background thread calls it again on refresh (the "subsequent"
+      -- call), which our helper makes fail with the supplied exception.
+      res <- try @AuthError $ do
+                _ <- fetchAuthInBackground menv
+                threadDelay 700000 -- give the background thread time to run (0.7s)
```

Comment:
Pull this out and make it a constant in `Test.Amazonka.Auth.Background`, once you create that. I think we can probably make it even shorter.